modules/telemetry/telemetry/tests/log_tests.cpp

```diff
@@ -72,118 +114,79 @@ TEST(log, Escapes) {
   }
 }
 
-/// @brief Custom sink that will write the log to a string.
-class MockLogSink final : public ht::ILogSink {
-public:
-  void send(const ht::LogEntry& s) override {
-    log_ += fmt::format("{}", s);
-    flag_.test_and_set();
-    flag_.notify_all();
-  }
-
-  [[nodiscard]] auto getLog() const -> const std::string& {
-    return log_;
-  }
-
-  void wait() const {
-    flag_.wait(false);
-  }
-
-private:
-  std::string log_;
-  std::atomic_flag flag_ = ATOMIC_FLAG_INIT;
-};
-
-TEST(log, sink) {
+TEST_F(LogTest, sink) {
   const int num = 123;
 
-  auto log_entry = ht::LogEntry{ heph::ERROR, "test another great message" }
-                   << ht::Field{ .key = "num", .value = num };
-  auto mock_sink = std::make_unique<MockLogSink>();
-  const MockLogSink* sink_ptr = mock_sink.get();
-  heph::telemetry::registerLogSink(std::move(mock_sink));
+  // clang-format off
+  const auto loc = std::source_location::current(); auto log_entry = ht::LogEntry{ heph::ERROR, "test another great message" } << ht::Field{ .key = "num", .value = num };
+  // clang-format on
+
   ht::internal::log(std::move(log_entry));
-  sink_ptr->wait();
+  fmt::print("{}", sink_ptr->getLog(loc));
   {
     std::stringstream ss;
     ss << "num=" << num;
-    EXPECT_TRUE(sink_ptr->getLog().find(ss.str()) != std::string::npos);
+    EXPECT_TRUE(sink_ptr->getLog(loc).find(ss.str()) != std::string::npos);
```

Comment:
```c++
ht::internal::log(heph::ERROR, "test another great message", "num", num);
auto log_line = sink_ptr->getLog();
{
    std::stringstream ss;
    ss << "num=" << num;
    EXPECT_TRUE(log_line.find(ss.str()) != std::string::npos);
}

```

Reply:
this would make the LogTest.sink identical to LogTest.log
The reason behind that test is to test the inner logging mechanics (logging LogEntry) independantly from the varidiac heph::log interface. makes sense or shall i remove the test?